src/tests/ftest/deployment/critical_integration.py

```diff
@@ -0,0 +1,110 @@
+                                      " 'echo hello'".format(','.join(self.hostlist_servers)))
+            try:
+                out = run_command(daos_server_cmd)
+                daos_server_version_list.append(out.stdout.split(b' ')[3])
+                if check_remote_root_access:
+                    run_command(remote_root_access)
+                IorTestBase._execute_command(self, command_for_inter_node, hosts=[host])
+            except (DaosTestError, CommandFailure) as error:
+                self.log.error("Error: %s", error)
+                failed_nodes.append(host)
+        if failed_nodes:
+            self.fail("SSH check failed on the following nodes.\n {}".format(failed_nodes))
+
+        for host in self.hostlist_clients:
+            dmg_version_cmd = ("ssh -oNumberOfPasswordPrompts=0 {}"
+                               " 'dmg version -i'".format(host))
+
+            try:
+                out = run_command(dmg_version_cmd)
+                dmg_version_list.append(out.stdout.split(b' ')[2])
+            except DaosTestError as error:
+                self.log.error("Error: %s", error)
+                failed_nodes.append(host)
+        if failed_nodes:
+            self.fail("SSH check for client nodes failed.\n {}".format(failed_nodes))
+
+        result_daos_server = (daos_server_version_list.count(daos_server_version_list[0])
+                              == len(daos_server_version_list))
+        result_dmg = dmg_version_list.count(dmg_version_list[0]) == len(dmg_version_list)
+        result_client_server = daos_server_version_list[0][1:] == dmg_version_list[0]
+
+        if (result_daos_server and result_dmg and result_client_server):
+            self.log.info("All servers have same daos version")
+            self.log.info("All clients have same daos version")
+            self.log.info("Servers and Clients have same daos version")
+            self.log.info("Clients: %s", dmg_version_list)
+            self.log.info("servers: %s", daos_server_version_list)
+        else:
+            self.log.info("Not all servers and clients have same daos version")
+            self.log.info("Clients: %s", dmg_version_list)
+            self.log.info("servers: %s", daos_server_version_list)
+            self.fail()
+
+    def test_ras(self):
+        """
+        Test Description: Verify RAS event on all server nodes from testrunner.
+                          Verify network scan and storage scan for server nodes.
+        :avocado: tags=all,deployment,full_regression
+        :avocado: tags=hw,large
+        :avocado: tags=criticalintegration,ras
+        """
+
+        dmg = self.get_dmg_command()
+        rank_list = self.server_managers[0].get_host_ranks(self.hostlist_servers)
+        print("rank_list: {}".format(rank_list))
+
+        for rank in rank_list:
+            dmg.system_stop(ranks=rank)
+            time.sleep(5)
+            dmg.system_start(ranks=rank)
+            time.sleep(5)
```

Comment:
Something else I'm noticing is that this won't scale very well. For example, 128 dual-engine servers:
128 * 2 = 256 ranks
256 * 10s = 2560s = 42.7 minutes
Maybe that's okay if we only ever run one time, but something to keep in mind.

Reply:
This test is intended to be run on per cabinet basis, hence it will be around 16 nodes (if there is none defected on that cabinet) hence 16*2*10s= ~5.3 mints. But if you have a better solution feel free to let me know. I want to refrain from stopping multiple ranks at the same time.

Reply:
Two improvements I can think of:
1. Stop all/multiple ranks before sleeping - but you don't want to do this.
2. Replace sleep with a deterministic way to know stop/start is in fact finished - I went through a similar scenario recently where I couldn't find a reliable way.

But if this is only for 16*2 engines, the runtime is okay.